inc/saitam.h

```diff
@@ -868,29 +1012,2440 @@ typedef sai_status_t (*sai_get_tam_transporter_attribute_fn)(
+{
+
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_MATH_FUNC_ATTR_START,
+
+    /**
+     * @brief Type of math function
+     * @type sai_tam_tel_math_func_type_t
+     * @flags CREATE_AND_SET
+     * @default SAI_TAM_TEL_MATH_FUNC_TYPE_NONE
+     */
+    SAI_TAM_MATH_FUNC_ATTR_TAM_TEL_MATH_FUNC_TYPE = SAI_TAM_MATH_FUNC_ATTR_START,
+
+    /**
+     * @brief TAM Object
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM
+     */
+    SAI_TAM_MATH_FUNC_ATTR_TAM_ID,
+
+    /**
+     * @brief Add Math func attributes below
+     */
+
+    /**
+     * @brief End of Attributes
+     */
+    SAI_TAM_MATH_FUNC_ATTR_END,
+
+    /** Custom range base value */
+    SAI_TAM_MATH_FUNC_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_TAM_MATH_FUNC_ATTR_CUSTOM_RANGE_END
+
+} sai_tam_math_func_attr_t;
+
+/**
+ * @brief Create and return a match function object
+ * @param[out] tam_math_func_id Object id for math function
+ * @param[in] switch_id Switch object id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_tam_math_func_fn)(
+        _Out_ sai_object_id_t *tam_math_func_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Deletes a specified Match function object
+ *
+ * @param[in] tam_math_func_id Object id for math function
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_tam_math_func_fn)(
+        _In_ sai_object_id_t tam_math_func_id);
+
+/**
+ * @brief Get values for specified Math function attributes
+ *
+ * @param[in] tam_math_func_id Object id for math function
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_tam_math_func_attribute_fn)(
+        _In_ sai_object_id_t tam_math_func_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Set value for specified Math function attribute
+ *
+ * @param[in] tam_math_func_id Object id for math function
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_tam_math_func_attribute_fn)(
+        _In_ sai_object_id_t tam_math_func_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Event Threshold Attributes
+ */
+typedef enum _sai_tam_event_threshold_attr_t
+{
+
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_EVENT_THRESHOLD_ATTR_START,
+
+    /**
+     * @brief High water mark %
+     *
+     * @type sai_uint32_t
```

Comment:
Is this value between 0-100?  should we use sai_uint8_t ?

Reply:
will change it